keras/layers/pooling.py

```diff
@@ -462,8 +462,22 @@ class GlobalAveragePooling1D(_GlobalPooling1D):
         `(batch_size, features)`
     """
 
-    def call(self, inputs):
-        return K.mean(inputs, axis=1)
+    def __init__(self, **kwargs):
+        super(GlobalAveragePooling1D, self).__init__(**kwargs)
+        self.supports_masking = True
+
+    def call(self, inputs, mask=None):
+        if mask is not None:
+            mask = K.cast(mask, K.floatx())
+            mask = K.repeat(mask, K.int_shape(inputs)[-1])
```

Comment:
You should use a broadcast rather than a repeat, for performance reasons

Reply:
Good point!